src/Microsoft.ComponentDetection.Contracts/TypedComponent/VcpkgComponent.cs

```diff
@@ -42,14 +42,19 @@ public override string Id
     {
         get
         {
-            if (this.PortVersion > 0)
+            var componentLocationPrefix = string.Empty;
+            if (!string.IsNullOrWhiteSpace(this.DownloadLocation) && !this.DownloadLocation.Trim().Equals("NONE", System.StringComparison.InvariantCultureIgnoreCase))
             {
-                return $"{this.Name} {this.Version}#{this.PortVersion} - {this.Type}";
+                componentLocationPrefix = $"{this.DownloadLocation} : ";
```

Comment:
The main change is including the downloadlocation of the component as prefix, the rest of the string stayed the same.